src/test/java/org/mskcc/cbio/oncokb/web/rest/UsageAnalysisControllerIT.java

```diff
@@ -0,0 +1,931 @@
+    String url = "/api/usage/summary/users";
+    String expected = gson.toJson(data.files.get(url));
+    restMockMvc
+      .perform(get(url))
+      .andExpect(status().isOk())
+      .andExpect(
+        content().contentTypeCompatibleWith(MediaType.APPLICATION_JSON)
+      )
+      .andExpect(content().json(expected, true));
+  }
+
+  @Test
+  public void shouldGetUsageSummaryUsersByCompanyId() throws Exception {
+    String url =
+      "/api/usage/summary/users?companyId=" +
+      data.userDtos.get(0).getCompany().getId();
+    String expected = gson.toJson(data.files.get(url));
+    restMockMvc
+      .perform(get(url))
+      .andExpect(status().isOk())
+      .andExpect(
+        content().contentTypeCompatibleWith(MediaType.APPLICATION_JSON)
+      )
+      .andExpect(content().json(expected, true));
+  }
+
+  @Test
+  public void shouldGetUsageUsersById() throws Exception {
+    String url = "/api/usage/users/" + data.users.get(0).getId();
+    String expected = gson.toJson(data.files.get(url));
+    restMockMvc
+      .perform(get(url))
+      .andExpect(status().isOk())
+      .andExpect(
+        content().contentTypeCompatibleWith(MediaType.APPLICATION_JSON)
+      )
+      .andExpect(content().json(expected, true));
+  }
+
+  private static class MockS3Data {
+
+    private static class KeyValuePair<Key, Value> {
+      public final Key key;
+      public final Value value;
+
+      public KeyValuePair(Key key, Value value) {
+        this.key = key;
+        this.value = value;
+      }
+    }
+
+    private final String[] sampleOncokbEndpoints = {
+      "/api/v1/annotate/mutations/byProteinChange",
+      "/api/private/utils/numbers/main/",
+      "/api/v1/annotate/structuralVariants",
+      "/api/private/utils/numbers/levels/",
+      "/api/v1/annotate/copyNumberAlterations",
+      "/api/private/search/typeahead",
+    };
+    private final String[] fakeCompanies = { "FakeCo", "DemoLLC" };
+    private final Random random;
+
+    private final JsonObject files = new JsonObject();
+
+    private final List<User> users = new ArrayList<>();
+    private final List<UserDTO> userDtos = new ArrayList<>();
+
+    public MockS3Data(int seed) throws Exception {
+      this(seed, new String[] { "doej", "smithj" });
+    }
+
+    public MockS3Data(int seed, String[] userNames) throws Exception {
+      random = new Random(seed);
+      addUsers(userNames);
+    }
+
+    private String getNextOncokbEndpoint() {
+      return sampleOncokbEndpoints[Math.abs(
+          random.nextInt() % sampleOncokbEndpoints.length
+        )];
+    }
+
+    /**
+     * Get value from files JSON object based on the JSON path created by the
+     * {@code keys}. If an object is missing in the JSON path then it's created.
+     * @param keys The keys in JSON path
+     * @return The resulting {@code KeyValuePair}
+     */
+    private KeyValuePair<String, JsonObject> fetchJsonObjectFromFilesObject(
```

Comment:
Two fetchJsonObjectFromFilesObject methods have the same name and same return but produce completely different JSONObject, one with last index included and set the initial value and one without last index. 
Can we maybe update the comment and even the method name?

Reply:
I'll remove the logic to set the initial value to avoid confusion.